src/Http/Routing/src/ValidationEndpointFilterFactory.cs

```diff
@@ -73,7 +77,8 @@ public static EndpointFilterDelegate Create(EndpointFilterFactoryContext context
                     validateContext = new ValidateContext
                     {
                         ValidationOptions = options,
-                        ValidationContext = validationContext
+                        ValidationContext = validationContext,
+                        SerializerOptions = jsonOptions?.SerializerOptions
```

Comment:
Once we make SerializerOptions on the ValidateContext internal, we can undo these changes to the ValidationEndpointFilterFactory.